### PR TITLE
refactor(components): [affix] use vueuse utils

### PR DESCRIPTION
**Pull request by sxzz**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/affix/src/affix.vue [hunk 1]
```diff
@@ -26,101 +31,92 @@ const ns = useNamespace('affix')
 const target = shallowRef<HTMLElement>()
 const root = shallowRef<HTMLDivElement>()
 const scrollContainer = shallowRef<HTMLElement | Window>()
+const { height: windowHeight } = useWindowSize()
+const rootRect = useElementBounding(root)
```
Comment (JeremyWuuuuu) [inline packages/components/affix/src/affix.vue:35]: Can we do this?

```ts
const { height: rootHeight, width: rootWidth } = useElementBounding(root)
```

File: packages/components/affix/src/affix.vue [hunk 2]
```diff
@@ -26,101 +31,92 @@ const ns = useNamespace('affix')
...
   emit('scroll', {
     scrollTop: scrollTop.value,
     fixed: fixed.value,
   })
 }
 
-watch(fixed, (fixed) => {
-  emit('change', fixed)
-})
+watch(fixed, () => emit('change', fixed.value))
```
Comment (JeremyWuuuuu) [inline packages/components/affix/src/affix.vue:103]:
```suggestion
watch(fixed, (val) => emit('change', val))
```

File: packages/components/affix/src/affix.vue [hunk 3]
```diff
@@ -26,101 +31,92 @@ const ns = useNamespace('affix')
...
   }
   scrollContainer.value = getScrollContainer(root.value!, true)
+  rootRect.update()
 })
 
-useEventListener(scrollContainer, 'scroll', onScroll)
-useResizeObserver(root, () => update())
-useResizeObserver(target, () => update())
+useEventListener(scrollContainer, 'scroll', handleScroll)
+watchEffect(update)
```
Comment (JeremyWuuuuu) [inline packages/components/affix/src/affix.vue:119]: Do we have perf concern here? Since every reactive value is tracked by `watchEffect`

Comment (sxzz) [reply packages/components/affix/src/affix.vue:119]: I think it's acceptable...

**Review by jw-foss — APPROVED** [review]

**Pull request merged**